src/OpenTelemetry.Extensions/Internal/BaggageLogRecordProcessor.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using OpenTelemetry.Logs;
+
+namespace OpenTelemetry.Extensions.Internal;
+
+internal sealed class BaggageLogRecordProcessor : BaseProcessor<LogRecord>
+{
+    private readonly Predicate<string> baggageKeyPredicate;
+
+    public BaggageLogRecordProcessor(Predicate<string> baggageKeyPredicate)
+    {
+        this.baggageKeyPredicate = baggageKeyPredicate ?? throw new ArgumentNullException(nameof(baggageKeyPredicate));
+    }
+
+    public static Predicate<string> AllowAllBaggageKeys => (_) => true;
+
+    public override void OnEnd(LogRecord data)
+    {
+        if (data != null && Baggage.Current.Count > 0)
```

Comment:
There is some cost to access `Baggage.Current` so it might be worth setting to a local once and then operating on that instead of `Baggage.Current` eg:

```csharp
    public override void OnEnd(LogRecord data)
    {
       var baggage = Baggage.Current;
       if (data != null && baggage.Count > 0)
       {
            var capacity = (data.Attributes?.Count ?? 0) + baggage.Count;
            var attributes = new List<KeyValuePair<string, object?>>(capacity);

            foreach (var entry in baggage)
            {
               // not shown
            }

            // not shown
       }
    }
```